paraatm/io/weather_cube_ET.py

```diff
@@ -0,0 +1,361 @@
+            weather_tensor.append(weather_v)
+
+        print("Total time for one trajectory is: ", time.time() - start)
+
+        # save data
+        if save_area is True:
+            np.save('weather_area', weather_tensor)
+        if save_point is True:
+            np.save('weather_point', point_t)
+
+
+def lat2y(a):
+    """
+    Calculate the WGS84 latitude into latitude under Mercator Projection
+
+    Parameters
+    ----------
+    a: Latitude in WGS84.
+
+    Returns
+    -------
+    Latitude in Mercator
+    """
+    Radius = 6378137.0  # Radius of Earth
+    return math.log(math.tan(math.pi / 4 + math.radians(a) / 2)) * Radius
+
+
+def lot2x(a):
+    """
+    Calculate the WGS84 longitude into longitude under Mercator Projection
+
+    Parameters
+    ----------
+    a: longitude in WGS84.
+
+    Returns
+    -------
+    longitude in Mercator
+    """
+    Radius = 6378137.0  # Radius of Earth
+    return math.radians(a) * Radius
+
+
+def find_nearest_value(array, num):
+    """
+    Find the nearest value to a given a number in a pre-defined array
+
+    Parameters
+    ----------
+    array: float, int
+    The reference array.
+
+    Returns
+    -------
+    num: float
+    The number we are interested.
+    """
+    nearest_val = array[abs(array - num) == abs(array - num).min()]
+    return nearest_val
+
+
+def check_convective_weather_files(weather_path, unix_time):
+    """
+    Find the closest ET weather files corresponding to the time of the current location
+
+    Parameters
+    ----------
+    weather_path: str
+    The path to the weather data.
+    unix_time: int
+    The unix time of the current location
+
+    Returns
+    -------
+    file_path: str
+    The path to the closest ET weather file at the current location and the current time
+    """
+    pin = datetime.datetime.utcfromtimestamp(int(float(unix_time))).strftime('%Y%m%d %H%M%S')  # time handle to check CIWS database
+    array = np.asarray([0, 230, 500, 730,
+                        1000, 1230, 1500, 1730,
+                        2000, 2230, 2500, 2730,
+                        3000, 3230, 3500, 3730,
+                        4000, 4230, 4500, 4730,
+                        5000, 5230, 5500, 5730])
+
+    nearest_value = int(find_nearest_value(array, 0.001+np.asarray([int(eliminate_zeros(pin[-4:]))])))  # find the closest time for downloading data from CIWS
+    nearest_value = make_up_zeros(str(nearest_value))  # make up zeros for 0 230 500 730
+
+    filename = pin[:8] + "ET/ciws.EchoTop." + pin[:8] + "T" + str(pin[-6:-4]) + nearest_value + "Z.nc"
+
+    file_path = weather_path + filename
+    return file_path
+
+
+def eliminate_zeros(num):
+    """
+    clear the zeros in the output of the unix to time transformation.
+
+    Parameters
+    ----------
+    num: float
+    A four digit number represents the minute and second time.
+
+    Returns
+    -------
+    num: float

```

Comment:
Same as above, isn't this really a `str`?

Reply:
Yes, this is str. Sorry for the misleading information.